### ISSUE TITLE
Stopped working in Nightly

### ISSUE THREAD

**Issue by nazar-pc** [gomita/firegestures#168]
This is the third extension that stopped working entirely in latest Nightlies.
Did they start removing APIs?
Doesn't work at all, worked yesterday, should be very straightforward to reproduce.

**Comment by tps125** [comment]
I can concur that it is not working for me on nightly either.

**Comment by nazar-pc** [comment]
I've just realized Firefox 57 is the first version that will support ONLY WebExtensions: https://blog.mozilla.org/addons/2017/02/16/the-road-to-firefox-57-compatibility-milestones/

So yes, indeed they're dropping functionality already. And while this extension could possibly be rewritten as WebExtension, many others probably will not be able to do this.

**Comment by Matth7878** [comment]
Can confirm, hope this extension will be converted soon.

Meanwhile webext alternative already exists with limitations inherent to them (not working on special pages and on addons.mozilla.org) but it's the only way to go. We will have to hope in long term these limitations will disappear : see [bugzilla1310082](https://bugzilla.mozilla.org/show_bug.cgi?id=1310082) which sadly is marked won't fix...

**Comment by nazar-pc** [comment]
WebExtensions are not working properly because of https://bugzilla.mozilla.org/show_bug.cgi?id=1360278

Also I'm not expecting this extension to be rewritten to be honest. If linked issue will be fixed I'll use Gesturefy.

**Comment by Matth7878** [comment]
@nazar-pc yes, context menu on mouse up sucks for linux and mac users. :-/
Foxy Gestures is also a nice replacement. I will test Gesturefy.

**Comment by Alex0007** [comment]
To all Linux users with Nightly 57+ I recommend to use [this extension](https://addons.mozilla.org/en-US/firefox/addon/smartup/?src=search). Because it prevents right click menu showing when doing gesture

**Comment by nazar-pc** [comment]
It simply disabled context menu and makes it available on doubleclick. Not ideal, but at least something.
Better push Firefox developers on the issue I've posted earlier to get proper fix in the platform.

**Comment by shinji2009** [comment]
+1 to extention update. hope and wait

**Comment by sbwtw** [comment]
+1

**Comment by kafene** [comment]
@nazar-pc 57 nightly (and supposedly Developer Edition once it's released) do still support legacy extensions like Firegestures, if you toggle some flags in `about:config`, I think it's just `extensions.legacy.enabled` and `extensions.allow-non-mpc-extensions`. These should be around for some time as Mozilla won't completely remove support for legacy extensions -even some of their "system add-ons" like the screenshots one and require legacy add-on capabilities.

**Comment by nazar-pc** [comment]
@kafene, theoretically yes. Practically some extensions break irreversibly anyway:(

**Comment by cswhatsgit** [comment]
@kafene Thanks for the tip. I'll give it a try when 57 stable rolls out, if any of my core addons break, I'll roll back to 56 and stay that way for the foreseeable future =(

**Comment by kukjevov** [comment]
Hi i have tried your workaround, but i have both `extensions.legacy.enabled` and `extensions.allow-non-mpc-extensions` set to `true` but this extension is not working in FF 57, 

I can see it now in Extensions page, it is enabled, but it is not working. Anyone with this problem?

**Comment by nazar-pc** [comment]
Forget about getting this working ever again, it is not going to happen

**Comment by kukjevov** [comment]
Ok, that means that this extensions is not going to work as it is or its development is dead and i should look for alternative (which i have already done).

**Comment by nazar-pc** [comment]
Not sure it is dead, but considering that it will require substantial effort, author of the extension didn't reply in this thread yet and alternatives already exist, I'd not keep much hope.

**Comment by will76** [comment]
RIP best Firefox extension.

Currently using StrokesPlus application. All of the _alternatives_ are laggy.

**Comment by genodeftest** [comment]
According to the Firefox WebExtensions List, there are 3 possible replacements:
* [Foxy Gestures](https://addons.mozilla.org/en-US/firefox/addon/foxy-gestures/)
* [Gesturefy](https://addons.mozilla.org/en-US/firefox/addon/gesturefy/)
* [smartUp Gestures](https://addons.mozilla.org/en-US/firefox/addon/smartup/)

All of them have a major showstopper on MacOS X and Linux, https://bugzilla.mozilla.org/show_bug.cgi?id=1360278
